### ISSUE TITLE
Test ASM code on multiple platforms

### ISSUE THREAD

**Issue by zonyitoo** [zonyitoo/context-rs#8]
Right now I can only test it on Linux x86_64 and OS X x86_64, not sure the ASM could run in other platforms.

**Comment by polyfractal** [comment]
Any chance of adding NetBSD support?  I looked around, but frankly I have no idea what asm needs to be changed/modified to make it work on NetBSD.  But I'd be happy to compile and run tests, or if you can point me towards a resource to learn more, I could send a PR (maybe heh)

**Comment by zonyitoo** [comment]
Can it be compiled on NetBSD?

**Comment by polyfractal** [comment]
Not at the moment, no.  There is no conditional compilation target for `target_record_sp_limit()` and `target_get_sp_limit()` (just freebsd and dragonfly).  

I considered trying out one of the existing versions, but since the asm is different for freebsd and dragonfly...i'm not really confident netbsd would be the same?  But honestly I don't really know what this code is doing anyway :(

I suppose I can try both and see if the tests pass?

**Comment by zonyitoo** [comment]
Yes, please. As I don't know what's the differences between NetBSD and FreeBSD, DragonFly in those functions.

**Comment by loongs-zhang** [comment]
Please assign it to me, see https://github.com/dragon-zhang/context-rs/tree/dev_add_ci , I think it's almost perfect @zonyitoo
